cmd/kubeadm/app/phases/certs/certs.go

```diff
@@ -68,35 +70,82 @@ func CreatePKIAssets(cfg *kubeadmapi.MasterConfiguration, pkiPath string) (*x509
 
 	altNames.IPs = append(altNames.IPs, internalAPIServerVirtualIP)
 
-	caKey, caCert, err := newCertificateAuthority()
-	if err != nil {
-		return nil, fmt.Errorf("failure while creating CA keys and certificate [%v]", err)
-	}
+	// Try to load ca.crt and ca.key from the PKI directory
+	caCert, caKey, oneOfThemExists, err := pkiutil.TryLoadCertAndKeyFromDisk(pkiDir, kubeadmconstants.CACertAndKeyBaseName)
 
-	if err := writeKeysAndCert(pkiPath, "ca", caKey, caCert); err != nil {
-		return nil, fmt.Errorf("failure while saving CA keys and certificate [%v]", err)
-	}
-	fmt.Println("[certificates] Generated Certificate Authority key and certificate.")
+	if err == nil && caCert != nil && caKey != nil {
+		// The certificate and key already exist, skip the generation
+		if !caCert.IsCA {
+			return nil, fmt.Errorf("certificate and key existed but the certificate is not a certificate authority")
+		}
 
-	apiKey, apiCert, err := newServerKeyAndCert(caCert, caKey, altNames)
-	if err != nil {
-		return nil, fmt.Errorf("failure while creating API server keys and certificate [%v]", err)
+		fmt.Println("[certificates] Using the existing CA certificate and key.")
+	} else if !oneOfThemExists {
```

Comment:
Try to invert this after removing oneOfThemExists so that the error flow is indented an the normal flow is unindented.